apps/src/applab/applab.js

```diff
@@ -1196,20 +1178,18 @@ Applab.execute = function () {
     }
   }
 
-  if (isMakerEnabled(getStore().getState())) {
-    connectToMakerBoard()
-        .then(board => {
-          board.installOnInterpreter(codegen, Applab.JSInterpreter);
-          makerCommands.injectBoardController(board);
-          board.once('disconnect', () => studioApp.resetButtonClick());
-          makerBoard = board;
-        })
-        .catch(error => studioApp.displayPlayspaceAlert('error',
-            <div>{`Board connection error: ${error}`}</div>))
-        .then(Applab.beginVisualizationRun);
-  } else {
-    Applab.beginVisualizationRun();
-  }
+    makerToolkit.connect({
+      interpreter: Applab.JSInterpreter,
+      onDisconnect: () => studioApp.resetButtonClick(),
+    })
+        .then(Applab.beginVisualizationRun)
+        .catch(error => {
+          // Don't just throw any error away, but squelch errors that we already
+          // handle gracefully (like early disconnect or a missing board).
+          if (!(error instanceof makerToolkit.MakerError)) {
+            Applab.log(error);
```

Comment:
Okay, the unhandled promise rejection seems to preserve the error just fine:
![image](https://cloud.githubusercontent.com/assets/1615761/24063290/4bad78a6-0b1c-11e7-870d-160c7d831694.png)
Chrome dev tools break on the unhandled rejection correctly:
![image](https://cloud.githubusercontent.com/assets/1615761/24063332/73dfce3c-0b1c-11e7-9da0-052b78a7a62d.png)
And the error object is _somewhat_ inspectable.  You don't get much of a stack, but then we rarely do with async logic.
![image](https://cloud.githubusercontent.com/assets/1615761/24063370/96b01818-0b1c-11e7-987c-c339d9e1139f.png)